### ISSUE TITLE
problems with ulpi-phy and usb-nop-xceiv in xilinx-v2015.3

### ISSUE THREAD

**Issue by pavel-demin** [Xilinx/linux-xlnx#68]
Hello,

Here are two kernel panic dumps that I get when trying to enable USB PHY. I tried both ulpi-phy and usb-nop-xceiv without much success.

In xilinx-v2015.2, usb-nop-xceiv works without any problem.

Best regards,

Pavel

kernel panic with ulpi-phy:
```
Unable to handle kernel NULL pointer dereference at virtual address 00000000
pgd = c0004000
[00000000] *pgd=00000000
Internal error: Oops - BUG: 5 [#1] PREEMPT SMP ARM
Modules linked in:
CPU: 0 PID: 1 Comm: swapper/0 Not tainted 4.0.0-xilinx #1
Hardware name: Xilinx Zynq Platform
task: dd42b9c0 ti: dd44a000 task.ti: dd44a000
PC is at ulpi_phy_probe+0x38/0xf8
LR is at ulpi_phy_probe+0x38/0xf8
pc : [<c0392008>]    lr : [<c0392008>]    psr: 60000113
sp : dd44be70  ip : 00000000  fp : 00000000
r10: c079aae4  r9 : c07f6980  r8 : 00000000
r7 : ddc2e660  r6 : dd521600  r5 : dd521610  r4 : dd551a10
r3 : c0391fd0  r2 : 00000000  r1 : 00000200  r0 : 00000000
Flags: nZCv  IRQs on  FIQs on  Mode SVC_32  ISA ARM  Segment kernel
Control: 18c5387d  Table: 0000404a  DAC: 00000015
Process swapper/0 (pid: 1, stack limit = 0xdd44a210)
Stack: (0xdd44be70 to 0xdd44c000)
be60:                                     c0391fd0 ffffffed dd521610 c07e8460
be80: c07e8460 c0288324 dd521610 c081e020 00000000 c0286edc dd521610 dd521644
bea0: c07e8460 c07bfb70 c07ad560 c02870c8 00000000 c07e8460 c0287060 c02857fc
bec0: dd49df5c dd4ca5b4 c07e8460 00000000 dd7bfa00 c0286794 c06de240 c06de241
bee0: 00000000 c07e8460 c0787954 00000000 c07ad560 c02875cc 00000000 dd667d00
bf00: c0787954 c0008900 c056f640 dd499c00 c056558c c07fd4f8 c079aa00 c00fce18
bf20: c06ab941 dd499e80 c07b0864 60000113 000000e6 ddfffdc9 00000000 c0035940
bf40: 000000e7 00000006 00000006 000000e6 c07b084c 00000006 000000e7 00000006
bf60: 000000e7 c07a09a4 c079aadc c07f6980 c07f6980 c076dd3c 00000006 00000006
bf80: c076d4fc dd44a000 00000000 c0556f04 00000000 00000000 00000000 00000000
bfa0: 00000000 c0556f0c 00000000 c000dd40 00000000 00000000 00000000 00000000
bfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
bfe0: 00000000 00000000 00000000 00000000 00000013 00000000 00000000 00000000
[<c0392008>] (ulpi_phy_probe) from [<c0288324>] (platform_drv_probe+0x48/0x98)
[<c0288324>] (platform_drv_probe) from [<c0286edc>] (driver_probe_device+0x98/0x1d8)
[<c0286edc>] (driver_probe_device) from [<c02870c8>] (__driver_attach+0x68/0x8c)
[<c02870c8>] (__driver_attach) from [<c02857fc>] (bus_for_each_dev+0x6c/0x90)
[<c02857fc>] (bus_for_each_dev) from [<c0286794>] (bus_add_driver+0xdc/0x1c4)
[<c0286794>] (bus_add_driver) from [<c02875cc>] (driver_register+0x8c/0xd0)
[<c02875cc>] (driver_register) from [<c0008900>] (do_one_initcall+0x100/0x180)
[<c0008900>] (do_one_initcall) from [<c076dd3c>] (kernel_init_freeable+0x110/0x1d4)
[<c076dd3c>] (kernel_init_freeable) from [<c0556f0c>] (kernel_init+0x8/0xe4)
[<c0556f0c>] (kernel_init) from [<c000dd40>] (ret_from_fork+0x14/0x34)
Code: e3a01c02 e3a02000 e1a00006 ebfbd75f (e8900006)
---[ end trace 76c87ff344a5606f ]---
Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b

CPU1: stopping
CPU: 1 PID: 0 Comm: swapper/1 Tainted: G      D         4.0.0-xilinx #1
Hardware name: Xilinx Zynq Platform
[<c001480c>] (unwind_backtrace) from [<c0010d4c>] (show_stack+0x10/0x14)
[<c0010d4c>] (show_stack) from [<c055a5d0>] (dump_stack+0x80/0xcc)
[<c055a5d0>] (dump_stack) from [<c00131e8>] (ipi_cpu_stop+0x3c/0x6c)
[<c00131e8>] (ipi_cpu_stop) from [<c00137d0>] (handle_IPI+0x64/0x84)
[<c00137d0>] (handle_IPI) from [<c00085e8>] (gic_handle_irq+0x54/0x5c)
[<c00085e8>] (gic_handle_irq) from [<c0011700>] (__irq_svc+0x40/0x74)
Exception stack(0xdd469f90 to 0xdd469fd8)
9f80:                                     00000001 00000000 00000000 c001c980
9fa0: 00000000 dd468000 00000000 c07f6c9c dd469fe0 00000001 00000000 c0562824
9fc0: 00000001 dd469fd8 c000e7d4 c000e7d8 60000113 ffffffff
[<c0011700>] (__irq_svc) from [<c000e7d8>] (arch_cpu_idle+0x2c/0x38)
[<c000e7d8>] (arch_cpu_idle) from [<c004911c>] (cpu_startup_entry+0x234/0x268)
[<c004911c>] (cpu_startup_entry) from [<00008684>] (0x8684)
---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b
```
kernel panic with usb-nop-xceiv:
```
Unable to handle kernel NULL pointer dereference at virtual address 00000000
pgd = c0004000
[00000000] *pgd=00000000
Internal error: Oops - BUG: 80000005 [#1] PREEMPT SMP ARM
Modules linked in:
CPU: 1 PID: 1 Comm: swapper/0 Not tainted 4.0.0-xilinx #1
Hardware name: Xilinx Zynq Platform
task: dd42b9c0 ti: dd44a000 task.ti: dd44a000
PC is at 0x0
LR is at ehci_ci_portpower+0x90/0xe8
pc : [<00000000>]    lr : [<c0396bac>]    psr: 20000113
sp : dd44b900  ip : 00000000  fp : de94a1c0
r10: 00000001  r9 : db9a8968  r8 : db9a8800
r7 : db9a0610  r6 : 00000001  r5 : dd4a7010  r4 : 00000000
r3 : 00000000  r2 : 00000000  r1 : 00000001  r0 : dd793110
Flags: nzCv  IRQs on  FIQs on  Mode SVC_32  ISA ARM  Segment kernel
Control: 18c5387d  Table: 0000404a  DAC: 00000015
Process swapper/0 (pid: 1, stack limit = 0xdd44a210)
Stack: (0xdd44b900 to 0xdd44c000)
b900: c0396b1c de94a180 8c000000 00000001 00000000 c0386c20 00000000 db9a8800
b920: de94a180 00000001 00000000 db9a8a2c 80000113 c0388cd0 00000000 00000002
b940: db9a8968 00000001 de94a184 00000008 dd44b98c dd756900 db9a8800 00000000
b960: 00002303 c03882fc db9ad480 00000000 00000008 c0378380 db9ad480 00000000
b980: dd412ec8 00000001 00000023 00000003 dd412ec8 00000000 00000005 dd412ec8
b9a0: 00000000 db8b5100 000014e0 c01dcef0 dd412ec8 db80a5d8 000003e0 dd412ec8
b9c0: 20000113 dd43d968 00000000 dd42ba08 ddc1d880 000160e3 00000000 dd42b9c0
b9e0: ddc1d840 dd42b9c0 dd4f7a08 c0041288 ddc1d840 00000000 dd44ba2c 00000000
ba00: dd756900 dd44ba24 dd44ba54 000003e8 00000023 00000001 00000001 c0379844
ba20: dd42ba08 00000000 00000000 dd44ba2c dd44ba2c db9ad5c0 db9ad5c0 00000000
ba40: 80000100 db9a8400 00000003 c0379950 db9a8400 80000100 00000003 00000001
ba60: db9a0000 00000000 db9a0048 00000001 db9a8800 c03706a4 00000008 00000001
ba80: 00000000 00000000 000003e8 db9a0000 00000000 c037320c db9ad680 db9a0000
baa0: 0000000a 00000000 db9a8400 c03732f8 dd767e6c 60000113 00000001 db9a8800
bac0: dd44bad4 c055d04c 60000113 00000000 db9a0000 db9a8400 db9a0200 00000000
bae0: 00000001 db9a8800 00000001 c0375f10 00002900 00000000 db9ad680 0000000f
bb00: 00001388 db9a84cc 00000004 db9ada40 db9a0220 00002900 c05a7e90 00000000
bb20: 00000000 db9a0220 00000000 db9a8400 c07e67ec 00000000 db9a0200 c05a7e90
bb40: 00000050 c037d01c db9a0220 c0820060 00000000 c07e67ec 00000000 c07e6890
bb60: c082003c c0286edc 00000000 db9a0220 c028701c db9a8468 00000000 c02858b8
bb80: dd5aee70 dd79ddb8 db9a0220 c07e68a8 db9a0254 c0286e04 db9a0220 c07e68a8
bba0: db9a0220 c0286574 db9a0220 db9a0228 00000000 c0284cd0 db9a0220 00000003
bbc0: db9a8400 00000082 db9a8400 db9a0200 db9a0450 db9a0450 db9a8468 db9ad840
bbe0: 00000001 c037b6ec 00000001 00000000 00000000 00000000 00001388 c01042d8
bc00: db80ad58 db9a8800 00000000 00000004 db9a0450 00000000 db9ad844 db9a8468
bc20: db9ad844 00000001 c07e6754 db9a8400 00000001 00000000 c07e6f80 00000000
bc40: c07e6754 c082003c 00000000 c03838e0 c03838a4 db9a8468 c0820060 c0286edc
bc60: 00000000 db9a8468 c028701c db9a0610 00000000 c02858b8 dd5aee70 dd79dd38
bc80: db9a8468 c07e68a8 db9a849c c0286e04 db9a8468 c07e68a8 db9a8468 c0286574
bca0: db9a8468 db9a8470 00000000 c0284cd0 39383168 6000303a 00000000 00000083
bcc0: db9a8400 db9a8468 00000000 00000000 db9a88d8 00000001 db9a0610 c03740b0
bce0: 00001388 db9a8800 db9a0610 db9a8800 00000012 00000012 00000012 00000012
bd00: db9a8400 db9adb40 db9a88d8 db9a8800 00000012 db9a8400 00000000 db9a88d8
bd20: 00000001 db9a0610 00000000 c03777fc db9a8800 00000000 00000000 db9a8400
bd40: db9a8800 dd4a7010 00000000 db9a0600 00000000 00000000 c082003c c0396d10
bd60: c0396c14 dd4a7010 ffffffff db9a0610 db9a0600 c0393b7c db9a0600 00000001
bd80: 00000000 c03936a4 ffffffed db9a0610 c07ea518 c07ea518 00000000 c0288324
bda0: db9a0610 c0820060 00000000 c0286edc 00000000 db9a0610 c028701c dd521a10
bdc0: 00000000 c02858b8 dd49df70 dd7f6d38 db9a0610 c07c1b70 db9a0644 c0286e04
bde0: db9a0610 c07c1b70 db9a0610 c0286574 db9a0610 db9a0618 00000000 c0284cd0
be00: db9a0610 db88cf9c 00000000 00000002 db9a0600 0000001c 00000002 00000000
be20: 00000002 dd4cfac0 00000000 c0288100 e0002000 db9a0600 c07ea670 00000000
be40: 00000000 c03935a4 00000000 dd521a10 dd521a00 db9a9190 ffffffff 00000000
be60: c07ea670 c03980e8 ffffffed dd521a10 c07ea600 c07ea600 00000000 c07f89c0
be80: c079cb04 c0288324 dd521a10 c0820060 00000000 c0286edc dd521a10 dd521a44
bea0: c07ea600 c07c1b70 c07af560 c02870c8 00000000 c07ea600 c0287060 c02857fc
bec0: dd49df5c dd4ca734 c07ea600 00000000 dd7f6c00 c0286794 c06dfb74 c06dfb76
bee0: 00000000 c07ea600 c0789984 00000000 c07af560 c02875cc 00000000 db9a9980
bf00: c0789984 c0008900 c0570640 dd499c00 c056658c c07ff538 c079ca00 c00fce18
bf20: c06acd31 dd499e80 c07b2864 60000113 000000e6 ddfffdc9 00000000 c0035940
bf40: 000000e7 00000006 00000006 000000e6 c07b284c 00000006 000000e7 00000006
bf60: 000000e7 c07a29d0 c079cafc c07f89c0 c07f89c0 c076fd3c 00000006 00000006
bf80: c076f4fc dd44a000 00000000 c0557c10 00000000 00000000 00000000 00000000
bfa0: 00000000 c0557c18 00000000 c000dd40 00000000 00000000 00000000 00000000
bfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
bfe0: 00000000 00000000 00000000 00000000 00000013 00000000 00000000 00000000
[<c0396bac>] (ehci_ci_portpower) from [<c0386c20>] (ehci_port_power+0xac/0xb8)
[<c0386c20>] (ehci_port_power) from [<c0388cd0>] (ehci_hub_control+0x9d4/0xb90)
[<c0388cd0>] (ehci_hub_control) from [<c0378380>] (usb_hcd_submit_urb+0x52c/0x74c)
[<c0378380>] (usb_hcd_submit_urb) from [<c0379844>] (usb_start_wait_urb+0x44/0xb4)
[<c0379844>] (usb_start_wait_urb) from [<c0379950>] (usb_control_msg+0x9c/0xc4)
[<c0379950>] (usb_control_msg) from [<c03706a4>] (set_port_feature+0x40/0x48)
[<c03706a4>] (set_port_feature) from [<c037320c>] (hub_power_on+0x54/0x88)
[<c037320c>] (hub_power_on) from [<c03732f8>] (hub_activate+0xb8/0x4d0)
[<c03732f8>] (hub_activate) from [<c0375f10>] (hub_probe+0x43c/0x764)
[<c0375f10>] (hub_probe) from [<c037d01c>] (usb_probe_interface+0x140/0x1d8)
[<c037d01c>] (usb_probe_interface) from [<c0286edc>] (driver_probe_device+0x98/0x1d8)
[<c0286edc>] (driver_probe_device) from [<c02858b8>] (bus_for_each_drv+0x84/0x94)
[<c02858b8>] (bus_for_each_drv) from [<c0286e04>] (device_attach+0x64/0x88)
[<c0286e04>] (device_attach) from [<c0286574>] (bus_probe_device+0x28/0x80)
[<c0286574>] (bus_probe_device) from [<c0284cd0>] (device_add+0x3bc/0x4c8)
[<c0284cd0>] (device_add) from [<c037b6ec>] (usb_set_configuration+0x5c4/0x648)
[<c037b6ec>] (usb_set_configuration) from [<c03838e0>] (generic_probe+0x3c/0x74)
[<c03838e0>] (generic_probe) from [<c0286edc>] (driver_probe_device+0x98/0x1d8)
[<c0286edc>] (driver_probe_device) from [<c02858b8>] (bus_for_each_drv+0x84/0x94)
[<c02858b8>] (bus_for_each_drv) from [<c0286e04>] (device_attach+0x64/0x88)
[<c0286e04>] (device_attach) from [<c0286574>] (bus_probe_device+0x28/0x80)
[<c0286574>] (bus_probe_device) from [<c0284cd0>] (device_add+0x3bc/0x4c8)
[<c0284cd0>] (device_add) from [<c03740b0>] (usb_new_device+0x2f4/0x438)
[<c03740b0>] (usb_new_device) from [<c03777fc>] (usb_add_hcd+0x448/0x6ac)
[<c03777fc>] (usb_add_hcd) from [<c0396d10>] (host_start+0xfc/0x1a0)
[<c0396d10>] (host_start) from [<c0393b7c>] (ci_hdrc_probe+0x4d8/0x5dc)
[<c0393b7c>] (ci_hdrc_probe) from [<c0288324>] (platform_drv_probe+0x48/0x98)
[<c0288324>] (platform_drv_probe) from [<c0286edc>] (driver_probe_device+0x98/0x1d8)
[<c0286edc>] (driver_probe_device) from [<c02858b8>] (bus_for_each_drv+0x84/0x94)
[<c02858b8>] (bus_for_each_drv) from [<c0286e04>] (device_attach+0x64/0x88)
[<c0286e04>] (device_attach) from [<c0286574>] (bus_probe_device+0x28/0x80)
[<c0286574>] (bus_probe_device) from [<c0284cd0>] (device_add+0x3bc/0x4c8)
[<c0284cd0>] (device_add) from [<c0288100>] (platform_device_add+0x12c/0x1c4)
[<c0288100>] (platform_device_add) from [<c03935a4>] (ci_hdrc_add_device+0x198/0x1dc)
[<c03935a4>] (ci_hdrc_add_device) from [<c03980e8>] (ci_hdrc_usb2_probe+0x144/0x1dc)
[<c03980e8>] (ci_hdrc_usb2_probe) from [<c0288324>] (platform_drv_probe+0x48/0x98)
[<c0288324>] (platform_drv_probe) from [<c0286edc>] (driver_probe_device+0x98/0x1d8)
[<c0286edc>] (driver_probe_device) from [<c02870c8>] (__driver_attach+0x68/0x8c)
[<c02870c8>] (__driver_attach) from [<c02857fc>] (bus_for_each_dev+0x6c/0x90)
[<c02857fc>] (bus_for_each_dev) from [<c0286794>] (bus_add_driver+0xdc/0x1c4)
[<c0286794>] (bus_add_driver) from [<c02875cc>] (driver_register+0x8c/0xd0)
[<c02875cc>] (driver_register) from [<c0008900>] (do_one_initcall+0x100/0x180)
[<c0008900>] (do_one_initcall) from [<c076fd3c>] (kernel_init_freeable+0x110/0x1d4)
[<c076fd3c>] (kernel_init_freeable) from [<c0557c18>] (kernel_init+0x8/0xe4)
[<c0557c18>] (kernel_init) from [<c000dd40>] (ret_from_fork+0x14/0x34)
Code: bad PC value
---[ end trace 64f1fd5a10c1fb0a ]---
Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b

CPU0: stopping
CPU: 0 PID: 0 Comm: swapper/0 Tainted: G      D         4.0.0-xilinx #1
Hardware name: Xilinx Zynq Platform
[<c001480c>] (unwind_backtrace) from [<c0010d4c>] (show_stack+0x10/0x14)
[<c0010d4c>] (show_stack) from [<c055b2dc>] (dump_stack+0x80/0xcc)
[<c055b2dc>] (dump_stack) from [<c00131e8>] (ipi_cpu_stop+0x3c/0x6c)
[<c00131e8>] (ipi_cpu_stop) from [<c00137d0>] (handle_IPI+0x64/0x84)
[<c00137d0>] (handle_IPI) from [<c00085e8>] (gic_handle_irq+0x54/0x5c)
[<c00085e8>] (gic_handle_irq) from [<c0011700>] (__irq_svc+0x40/0x74)
Exception stack(0xc07a9f58 to 0xc07a9fa0)
9f40:                                                       00000001 00000000
9f60: 00000000 c001c980 00000000 c07a8000 00000000 c07ab440 c07a9fa8 00000001
9f80: 00000000 c0563824 00000001 c07a9fa0 c000e7d4 c000e7d8 60000113 ffffffff
[<c0011700>] (__irq_svc) from [<c000e7d8>] (arch_cpu_idle+0x2c/0x38)
[<c000e7d8>] (arch_cpu_idle) from [<c004911c>] (cpu_startup_entry+0x234/0x268)
[<c004911c>] (cpu_startup_entry) from [<c076fbcc>] (start_kernel+0x324/0x384)
[<c076fbcc>] (start_kernel) from [<00008074>] (0x8074)
---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x0000000b
```

**Comment by michalsimek** [comment]
Hi,

did you try the latest kernel from master branch?
If that is not working we have to replicate the problem on our side that's why please provide us information about board you use and about .config you use. Full bootlog will be also helpful.

Thanks,
Michal

**Comment by pavel-demin** [comment]
Hi Michal,

I've just tried the master branch. The kernel panic is still there.

Here is the information that you requested and some more:
 - the board is Red Pitaya
 - [.config file](http://googledrive.com/host/0B-t5klOOymMNfmJ0bFQzTVNXQ3RtWm5SQ2NGTE1hRUlTd3V2emdSNzN6d0pYamNILW83Wmc/ulpi/config)
 - [xilinx_zynq_defconfig](http://googledrive.com/host/0B-t5klOOymMNfmJ0bFQzTVNXQ3RtWm5SQ2NGTE1hRUlTd3V2emdSNzN6d0pYamNILW83Wmc/ulpi/xilinx_zynq_defconfig)
 - [full bootlog](http://googledrive.com/host/0B-t5klOOymMNfmJ0bFQzTVNXQ3RtWm5SQ2NGTE1hRUlTd3V2emdSNzN6d0pYamNILW83Wmc/ulpi/bootlog.txt)
 - [devicetree](http://googledrive.com/host/0B-t5klOOymMNfmJ0bFQzTVNXQ3RtWm5SQ2NGTE1hRUlTd3V2emdSNzN6d0pYamNILW83Wmc/ulpi/devicetree.dts)

Regards,

Pavel

**Comment by pavel-demin** [comment]
As a workaround, I've taken the previous version of [ci_hdrc_usb2.c](https://github.com/Xilinx/linux-xlnx/blob/24c498df1c70fa3b8b4df18f4424904bf014c709/drivers/usb/chipidea/ci_hdrc_usb2.c), [host.c](https://github.com/Xilinx/linux-xlnx/blob/24c498df1c70fa3b8b4df18f4424904bf014c709/drivers/usb/chipidea/host.c), [otg_fsm.c](https://github.com/Xilinx/linux-xlnx/blob/24c498df1c70fa3b8b4df18f4424904bf014c709/drivers/usb/chipidea/otg_fsm.c) and [chipidea.h](https://github.com/Xilinx/linux-xlnx/blob/24c498df1c70fa3b8b4df18f4424904bf014c709/include/linux/usb/chipidea.h). The previous version does not contain the VBUS control code and the driver is working without any problem.

To summarize:
 - current version (https://github.com/Xilinx/linux-xlnx/commit/7be1ce169687f4e3ccbfe87459dc20c12f558a3a) of the usb-nop-xceiv driver crashes during boot,
 - previous version (https://github.com/Xilinx/linux-xlnx/commit/24c498df1c70fa3b8b4df18f4424904bf014c709) of the usb-nop-xceiv driver works without any problem.

Now, I have the usb-nop-xceiv driver working in xilinx-v2015.3.

**Comment by michalsimek** [comment]
Please add phy in dts like this and let me know the result:

usb_phy0: phy0 {
                compatible = "ulpi-phy";
                #phy-cells = <0>;
                reg = <0xe0002000 0x1000>;
                view-port = <0x0170>;
                drv-vbus;
 };

I will add check in driver to look for phy is not NULL.
Sundeep

**Comment by pavel-demin** [comment]
With this new phy configuration the ulpi-phy driver works.

Out of curiosity, I tried to remove "reg = <0xe0002000 0x1000>;" and it crashed again.

**Comment by pavel-demin** [comment]
Correction to my previous comment: the ulpi-phy driver does not crash with the new phy configuration but it does not seem to work:
```
# lsusb
unable to initialize libusb: -99
```
The usb-nop-xceiv driver from xilinx-v2015.2 works:
```
# lsusb
Bus 001 Device 002: ID 0bda:8176 Realtek Semiconductor Corp. RTL8188CUS 802.11n WLAN Adapter
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

**Comment by michalsimek** [comment]
Strange. Can you reconfirm these two are also added in phy dts:
view-port = <0x0170>;
drv-vbus;

Also do you see ci_hrc.0 in sysfs (/sys/bus/platform/ci_hdrc.0/) ?

Sundeep

**Comment by pavel-demin** [comment]
Just copied and pasted the usb_phy0 configuration and recompiled my devicetree. The result is the same: no USB devices.
```
# lsusb
unable to initialize libusb: -99
```

The location of the ci_hrc.0 directory is slightly different on my system: 
```
# ls /sys/bus/platform/devices/ci_hdrc.0/
driver_override  modalias  power  subsystem  uevent
```

Here is how it looks like with the usb-nop-xceiv driver from xilinx-v2015.2:
```
# ls /sys/bus/platform/devices/ci_hdrc.0/
driver           modalias  power      uevent               usb1
driver_override  pools     subsystem  uframe_periodic_max
```

**Comment by michalsimek** [comment]
Can you please send your email address to git@xilinx.com? I have some inputs from Sundeep and it will be much easier for communication.

Thanks,
Michal

**Comment by pavel-demin** [comment]
Everything works after applying the latest patches.

Thanks a lot for fixing theses problems!

Closing this issue.

**Comment by PRJeanneret** [comment]
Hi, 

I'm experiencing *exactly* the same problem with the current "master", updated this morning. 

Best Regards

Paul

**Comment by pavel-demin** [comment]
Attached is a patch that I've received from the Xilinx developers and that solved the problem for me. 

I thought that it was already in the repository. But I don't see any commit corresponding to this patch.

[0001-usb-chipidea-Check-usb_phy-exists-before-using-it.txt](https://github.com/Xilinx/linux-xlnx/files/38925/0001-usb-chipidea-Check-usb_phy-exists-before-using-it.txt)

**Comment by pavel-demin** [comment]
Dear Xilinx developers,

Could you please commit to the repository the patch for `drivers/usb/chipidea/host.c` and `drivers/usb/chipidea/otg_fsm.c` that you sent me by e-mail and that solved this problem?

The patch is attached to my previous comment.

Best regards,

Pavel

**Comment by pavel-demin** [comment]
I've just noticed a new commit (https://github.com/Xilinx/linux-xlnx/commit/2f6e301e60806311cbf00145a1853e6f5ed95a12) with the patch mentioned in my previous comments.

Thanks again for fixing this problem.

Best regards,

Pavel
